-pthread flag missing

Building `bionic` branch on Arch GNU/Linux

```
/usr/bin/ld: ../../libGMock.a(gmock-all.cc.o): undefined reference to symbol 'pthread_key_delete@@GLIBC_2.2.5'
/usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [tests/unit/libusermetricsoutput/CMakeFiles/usermetricsoutput-unit-tests.dir/build.make:202: tests/unit/libusermetricsoutput/usermetricsoutput-unit-tests] Error 1
```

`-pthread` is missing in `target_link_libraries` in `CMakeLists.txt`

Same as https://github.com/ubports/url-dispatcher/issues/4

--- a/tests/unit/libusermetricsoutput/CMakeLists.txt
+++ b/tests/unit/libusermetricsoutput/CMakeLists.txt
@@ -17,6 +17,7 @@ add_executable(
 
 target_link_libraries(
 	usermetricsoutput-unit-tests
+	-pthread
 	usermetricsoutput-static
 	usermetrics-test-utils
 	${GSETTINGS_QT_LIBRARIES}

--- a/tests/unit/usermetricsservice/CMakeLists.txt
+++ b/tests/unit/usermetricsservice/CMakeLists.txt
@@ -12,6 +12,7 @@ add_executable(
 
 target_link_libraries(
 	usermetricsservice-unit-tests
+	-pthread
 	usermetricsservice
 	usermetrics-test-utils
 	${QTDBUSTEST_LIBRARIES}

